### ISSUE TITLE
[Dynamic Regions] br: br restore to multirocks cluster causing tikv full space

### ISSUE THREAD

**Issue by AkiraXie** [pingcap/tidb#44550]
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
br full restore to multirocks cluster

### 3. What did you see instead (Required)
[error="Cannot read http://minio-service:9000/longrun/full-0530/4/3347391867_49715_e1868a8b75a25f723512fff6a380f1770e6ade5da3e270f3211c422e6ccdff2a_1685528172509_default.sst into /var/lib/tikv/data/import/.temp/e577cfd7-abcf-4431-8c55-8be66bcc7317_76351_43_970_default.sst: No space left on device (os error 28)
![image](https://github.com/pingcap/tidb/assets/50363732/fdf109d5-605e-4b8d-a66e-3eb6a0ec6fbc)

### 4. What is your TiDB version? (Required)
Release Version: v7.1.0
Git Commit Hash: fd2314159d1a81c8b13e5d61e53dcd527cb35494
Git Branch: heads/refs/tags/v7.1.0
Go Version: go1.20.3
UTC Build Time: 2023-05-24 02:08:51
Race Enabled: false

**Comment by 3pointer** [comment]
I think I found the root cause of this issue.
In raftstore-v2, even after split empty regions. TiKV still [estimate region](https://github.com/tikv/tikv/blob/993eb2f610acfc4ee9e6c7b4f03eb63213f6d593/components/raftstore-v2/src/operation/command/admin/split.rs#L683) size by old_region_size/new_region_count. This will mislead PD to think the region has data (but actually it's empty) and report exceed store limit.

**Comment by 3pointer** [comment]
From the new reproduce log, we can find a cancelled operation of region: 1853


PD receive new split request and reports exceed store limit at 20:08:26.800 +08:00
```
[2023/07/07 20:08:26.731 +08:00] [INFO] [cluster_worker.go:241] ["region batch split, generate new regions"] [region-id=1063] [origin="id:1835 start_key:\"74800000000000E8FFF05F72FFFFFFFFFFFFFFFFFF0000000000FB\" end_key:\"74800000000000E8FFF25F698000000000FF000001FB00000000FB\" region_epoch:<conf_ver:31 version:208 > peers:<id:1836 store_id:21 > id:1837 start_key:\"74800000000000E8FFF25F698000000000FF000001FB00000000FB\" end_key:\"74800000000000E8FFF25F698000000000FF000002FB00000000FB\" region_epoch:<conf_ver:31 version:208 > peers:<id:1838 store_id:21 > id:1839 start_key:\"74800000000000E8FFF25F698000000000FF000002FB00000000FB\" end_key:\"74800000000000E8FFF25F698000000000FF000003FB00000000FB\" region_epoch:<conf_ver:31 version:208 > peers:<id:1840 store_id:21 > id:1841 start_key:\"74800000000000E8FFF25F698000000000FF000003FB00000000FB\" end_key:\"74800000000000E8FFF25F698000000000FF000004FB00000000FB\" region_epoch:<conf_ver:31 version:208 > peers:<id:1842 store_id:21 > id:1843 start_key:\"74800000000000E8FFF25F698000000000FF000004FB00000000FB\" end_key:\"74800000000000E8FFF25F698000000000FF000005FB00000000FB\" region_epoch:<conf_ver:31 version:208 > peers:<id:1844 store_id:21 > id:1845 start_key:\"74800000000000E8FFF25F698000000000FF000005FB00000000FB\" end_key:\"74800000000000E8FFF25F698000000000FF000006FB00000000FB\" region_epoch:<conf_ver:31 version:208 > peers:<id:1846 store_id:21 > id:1847 start_key:\"74800000000000E8FFF25F698000000000FF000006FB00000000FB\" end_key:\"74800000000000E8FFF25F72FFFFFFFFFFFFFFFFFF0000000000FB\" region_epoch:<conf_ver:31 version:208 > peers:<id:1848 store_id:21 > id:1849 start_key:\"74800000000000E8FFF25F72FFFFFFFFFFFFFFFFFF0000000000FB\" end_key:\"74800000000000E9FF075F698000000000FF000001FB00000000FB\" region_epoch:<conf_ver:31 version:208 > peers:<id:1850 store_id:21
 > id:1851 start_key:\"74800000000000E9FF075F698000000000FF000001FB00000000FB\" end_key:\"74800000000000E9FF075F698000000000FF000002FB00000000FB\" region_epoch:<conf_ver:31 version:208 > peers:<id:1852 store_id:21 > id:1853 start_key:\"74800000000000E9FF075F698000000000FF000002FB00000000FB\" end_key:\"74800000000000E9FF075F698000000000FF000003FB00000000FB\" region_epoch:<conf_ver:31 version:208 > peers:<id:1854 store_id:21 > id:1855 start_key:\"74800000000000E9FF075F698000000000FF000003FB00000000FB\" end_key:\"74800000000000E9FF075F698000000000FF000004FB00000000FB\" region_epoch:<conf_ver:31 version:208 > peers:<id:1856 store_id:21 > id:1857 start_key:\"74800000000000E9FF075F698000000000FF000004FB00000000FB\" end_key:\"74800000000000E9FF075F72FFFFFFFFFFFFFFFFFF0000000000FB\" region_epoch:<conf_ver:31 version:208 > peers:<id:1858 store_id:21 >"] [total=12]
[2023/07/07 20:08:26.800 +08:00] [INFO] [operator_controller.go:647] ["operator canceled"] [region-id=1853] [takes=0s] [operator="\"scatter-region {mv light peer: store [21] to [5]} (kind:admin,region,leader, region:1853(208, 31), createAt:2023-07-07 20:08:26.80091414 +0800 CST m=+322.472567274, startAt:0001-01-01 00:00:00 +0000 UTC, currentStep:0, size:6, steps:[add learner peer 1864 on store 5, use joint consensus, promote learner peer 1864 on store 5 to voter, demote voter peer 1854 on store 21 to learner, transfer leader from store 21 to store 5, leave joint state, promote learner peer 1864 on store 5 to voter, demote voter peer 1854 on store 21 to learner, remove peer on store 21], timeout:[18m0s])\""] [additional-info="{\"cancel-reason\":\"exceed store limit\"}"]
```

From PD log we can get region 1853 start key and end key is 
`[74800000000000E9FF075F698000000000FF000002FB00000000FB,  74800000000000E9FF075F698000000000FF000003FB00000000FB)` which table id is 59655.

From TiDB we can see this table is 
```
mysql> select * from information_schema.tables where tidb_table_id=59655;
+---------------+--------------+--------------------------------+------------+--------+---------+------------+------------+----------------+-------------+-----------------+--------------+-----------+----------------+---------------------+-------------+------------+-----------------+----------+----------------+---------------+---------------+---------------------------+--------------+----------------------------+
| TABLE_CATALOG | TABLE_SCHEMA | TABLE_NAME                     | TABLE_TYPE | ENGINE | VERSION | ROW_FORMAT | TABLE_ROWS | AVG_ROW_LENGTH | DATA_LENGTH | MAX_DATA_LENGTH | INDEX_LENGTH | DATA_FREE | AUTO_INCREMENT | CREATE_TIME         | UPDATE_TIME | CHECK_TIME | TABLE_COLLATION | CHECKSUM | CREATE_OPTIONS | TABLE_COMMENT | TIDB_TABLE_ID | TIDB_ROW_ID_SHARDING_INFO | TIDB_PK_TYPE | TIDB_PLACEMENT_POLICY_NAME |
+---------------+--------------+--------------------------------+------------+--------+---------+------------+------------+----------------+-------------+-----------------+--------------+-----------+----------------+---------------------+-------------+------------+-----------------+----------+----------------+---------------+---------------+---------------------------+--------------+----------------------------+
| def           | rg_betting_0 | unit2_14_game_bets_game_tag_30 | BASE TABLE | InnoDB |      10 | Compact    |      70605 |             50 |     3530250 |               0 |      1129680 |         0 |              1 | 2023-07-07 20:07:39 | NULL        | NULL       | utf8mb4_bin     |     NULL |                |               |         59655 | NOT_SHARDED(PK_IS_HANDLE) | CLUSTERED    | NULL                       |
+---------------+--------------+--------------------------------+------------+--------+---------+------------+------------+----------------+-------------+-----------------+--------------+-----------+----------------+---------------------+-------------+------------+-----------------+----------+----------------+---------------+---------------+---------------------------+--------------+----------------------------+
```

BR finished the restore of this table at 20:08:37.277 +08:00, so I believe there is no data in region. 
```
[2023/07/07 12:08:37.277 +00:00] [INFO] [client.go:1750] ["success in validate checksum"] [db=rg_betting_0] [table=unit2_14_game_bets_game_tag_30]
```

**Comment by 3pointer** [comment]
To solve this totally. I think we need https://github.com/tikv/pd/pull/6593#issuecomment-1627071878 if we cannot get correct size of new split region. PD then cannot rely on this size to decide the limitation.

**Comment by BornChanger** [comment]
@seiya-annie do we need to cherry-pick the fix to 7.1? It seems this issue only affects multi-rocksdb @3pointer
